src/float.rs

```diff
@@ -0,0 +1,310 @@
+    fn checked_add(&self, v: &Self) -> Option<Self> {
+        Some(self.value.add(v.value).into())
+    }
+}
+
+impl CheckedSub for Float64 {
+    fn checked_sub(&self, v: &Self) -> Option<Self> {
+        Some(self.value.sub(v.value).into())
+    }
+}
+
+impl CheckedMul for Float64 {
+    fn checked_mul(&self, v: &Self) -> Option<Self> {
+        Some(self.value.mul(v.value).into())
+    }
+}
+
+impl CheckedDiv for Float64 {
+    fn checked_div(&self, v: &Self) -> Option<Self> {
+        if v.is_zero() {
+            return None;
+        }
+        Some(self.value.div(v.value).into())
+    }
+}
+
+impl FromPrimitive for Float64 {
+    fn from_i64(n: i64) -> Option<Self> {
+        Some((n as f64).into())
+    }
+
+    fn from_u64(n: u64) -> Option<Self> {
+        Some((n as f64).into())
+    }
+}
+
+impl ToPrimitive for Float64 {
+    fn to_i64(&self) -> Option<i64> {
+        Some(self.value as i64)
+    }
+
+    fn to_u64(&self) -> Option<u64> {
+        Some(self.value as u64)
+    }
+}
+
+impl Signed for Float64 {
+    fn abs(&self) -> Self {
+        self.value.abs().into()
+    }
+
+    fn abs_sub(&self, other: &Self) -> Self {
+        (self.value - other.value).abs().into()
+    }
+
+    fn signum(&self) -> Self {
+        self.value.signum().into()
+    }
+
+    fn is_positive(&self) -> bool {
+        self.value.is_sign_positive()
+    }
+
+    fn is_negative(&self) -> bool {
+        self.value.is_sign_negative()
+    }
+}
+
+impl Neg for Float64 {
+    type Output = Self;
+
+    fn neg(self) -> Self::Output {
+        self.value.neg().into()
+    }
+}
+
+impl FromStr for Float64 {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        s.parse::<f64>().map(|v| v.into()).map_err(|_| ())
+    }
+}
+
+impl ToString for Float64 {
+    fn to_string(&self) -> String {
+        self.value.to_string()
+    }
+}
+
+impl LaTeXable for Float64 {
+    fn to_latex(&self) -> String {
+        self.value.to_latex()
+    }
+}
+
+impl GuiDisplayable for Float64 {
+    fn display_string(&self, locale: &Locale) -> String {
+        self.value.display_string(locale)
+    }
+
+    fn to_shape(&self, ctx: &Context, font_id: FontId, color: Color32) -> Shape {
+        self.value.to_shape(ctx, font_id, color)
+    }
+}
+
+impl LaTeXable for f64 {
+    fn to_latex(&self) -> String {
+        trim_trailing_zeros_float_str(&format!("{:.*}", FLOAT_STRING_PRECISION, self))
+    }
+}
+
+/// Trims trailing zeros from a float string.
+/// # Arguments
+/// * `s` - String to trim zeros from
+/// # Returns
+/// A string representing the same floating point number, but without trailing zeros
+/// # Examples
+/// ```rust
+/// # use crate::jp2gmd_lib::trim_trailing_zeros_float_str;
+/// assert_eq!(trim_trailing_zeros_float_str("10.0"), "10");
+/// assert_eq!(trim_trailing_zeros_float_str("123.450"), "123.45");
+/// assert_eq!(trim_trailing_zeros_float_str("1.0"), "1");
+/// ```
+pub fn trim_trailing_zeros_float_str(s: &str) -> String {
```

Comment:
`s` - very descriptive name @tudny

Reply:
does it really matter if this is an only argument and function name is already descriptive?

Reply:
It does, as the function doesn't only remove trailing zeros from the given string, but trims the zeros from the float number represented by this string. E.g. 
10.0 -> 10 (the dot is removed as well)
10 -> 10 (there is a trailing zero, that is not removed)